Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -200,7 +200,7 @@ private function verifyExpectedVersion(int $actualVersion, int $expectedVersion)
         if ($expectedVersion === ExpectedVersion::ANY) {
             return;
         }
-        if ($expectedVersion === $actualVersion) {
+        if ($expectedVersion === $actualVersion || ($expectedVersion === ExpectedVersion::STREAM_EXISTS && $actualVersion > -1)) {
```

Comment:
I'd prefer throwing a distinct exception if a `STREAM_EXISTS` or `NO_STREAM` expectation is given. In the case of `NO_STREAM` the underlying issue is probably concurrency, but in the case of `STREAM_EXISTS` is most likely a logic issue.

Reply:
Good point.. But I'd like to avoid adding yet another Exception, i.e. `@throws` annotation, i.e. `catch`. What do you think of changing the exception name to sth like `WrongExpectedNumberException`?

Reply:
...like https://github.com/FriendsOfOuro/geteventstore-php-core/blob/master/src/Exception/WrongExpectedVersionException.php

Reply:
@albe if you agree to this suggestion, I'll take care and push a follow-up

Reply:
Hm, not sure. That makes it even less tell what the exception means in relation to the underlying expectations set with "expectedVersion".

`ConcurrencyException` says: hey, there was some stuff going on concurrently - take a look! `WrongExpectedVersionException` only tells you something about an implementation detail went wrong. Now imagine that exception being thrown somewhere inside your app code - where you maybe don't even set that expected version yourself.

Maybe it's okay to leave it as is, in case you want to write to the stream, expecting it to just exist in any form, then you're doing pretty low-level stuff already, so
```php
try {
   ...->commit(...., ExpectedVersion::STREAM_EXIST);
} catch (ConcurrencyException $e) {
   throw new My\StreamDoesNotExist(...);
}
```
is okay to expect.

Reply:
yep, agreed. thanks!